### ISSUE TITLE
type annotations for IDA plugin

### ISSUE THREAD

**Issue by williballenthin** [mandiant/capa#1051]
the IDA plugin code does not have many type annotations because it was written before we migrated to using py3-style type annotations. for the same reasons we use type annotations elsewhere, we should consider adding type annotations to the IDA plugin source, too.

**Comment by williballenthin** [comment]
we could use something like [monkeytype](https://monkeytype.readthedocs.io/en/latest/index.html) to automatically generate the annotations given a runtime trace of the code.

**Comment by manasghandat** [comment]
Hi i would like to work on this issue

**Comment by mr-tz** [comment]
Great, I've assigned you. Thanks!

**Comment by manasghandat** [comment]
I was going through this when i encountered the issue that packages like `idac` and `idaapi` are not installed in my system. Do i need IDA Pro for it?

**Comment by williballenthin** [comment]
That’s correct, those imports are for the IDA API that’s only available within IDA Pro.

**Comment by manasghandat** [comment]
Can you please suggest me some issue where I wont need IDA Pro.

**Comment by mr-tz** [comment]
https://github.com/mandiant/capa/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22
